integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mv/DataMapListeners.scala

```diff
@@ -139,3 +148,130 @@ object LoadPostDataMapListener extends OperationEventListener {
     }
   }
 }
+
+/**
+ * Listeners to block operations like delete segment on id or by date on tables
+ * having an mv datamap or on mv datamap tables
+ */
+object DataMapDeleteSegmentPreListener extends OperationEventListener {
+  /**
+   * Called on a specified event occurrence
+   *
+   * @param event
+   * @param operationContext
+   */
+  override def onEvent(event: Event, operationContext: OperationContext): Unit = {
+    val carbonTable = event match {
+      case e: DeleteSegmentByIdPreEvent =>
+        e.asInstanceOf[DeleteSegmentByIdPreEvent].carbonTable
+      case e: DeleteSegmentByDatePreEvent =>
+        e.asInstanceOf[DeleteSegmentByDatePreEvent].carbonTable
+    }
+    if (null != carbonTable) {
+      if (CarbonTable.hasMVDataMap(carbonTable)) {
+        throw new UnsupportedOperationException(
+          "Delete segment operation is not supported on tables which have mv datamap")
```

Comment:
Don't put mv name here, just check for child tables and throw exception.

Reply:
done